modules/server.py

```diff
@@ -37,7 +45,6 @@ def process_request(self, request):
         self.processed_requests += 1
 
```

Comment:
Shouldn't we add memory to the heap when process the request?

Reply:
And it does, but it isn't explicitly implemented at this method *process_request*. It is implemented at the method *run*, of the class [**Request**](https://github.com/gcinterceptor/gci-simulator/blob/master/modules/clients.py#L14), that is called in *process_request*.

Reply:
Got it. I would place everything dealing with heap usage at the *same place*. The problem I see is that `heap.get` happens here and `heap.put` at request. I suggest consolidating those at one place.